crates/sui-move-build/src/lib.rs

```diff
@@ -158,23 +158,12 @@ impl BuildConfig {
         let print_diags_to_stderr = self.print_diags_to_stderr;
         let run_bytecode_verifier = self.run_bytecode_verifier;
         let resolution_graph = self.resolution_graph(&path)?;
-        let result = build_from_resolution_graph(
+        build_from_resolution_graph(
             path.clone(),
             resolution_graph,
             run_bytecode_verifier,
             print_diags_to_stderr,
-        );
-        if let Ok(ref compiled) = result {
-            compiled
-                .package
-                .compiled_package_info
-                .build_flags
-                .update_lock_file_toolchain_version(&path, env!("CARGO_PKG_VERSION").into())
-                .map_err(|e| SuiError::ModuleBuildFailure {
-                    error: format!("Failed to update Move.lock toolchain version: {e}"),
-                })?;
-        }
-        result
```

Comment:
Also updated in this PR: the toolchain update is lifted into `sui-move/src/build.rs` and `sui/src/client_commands` respectively. This ensures that `Move.lock` is updated for a `sui client publish` command even when `sui move build` has never been called before (indeed, this `build` function doesn't get called on `sui client publish`).